cockpit/leapp.html

```diff
@@ -875,28 +940,30 @@ <h3 class="pull-left">
         }
 
         function scan_ports(ip_addr, callback, ports=null, shallow=false) {
-            var scan_args = ['port-inspect', ip_addr];
-            if (ports !== null) {
-                scan_args.push('--range', ports);
+            // Make sure that shallow will be true or false only
+            if (shallow !== false) {
+                shallow = true
             }
-            if (shallow) {
-                scan_args.push('--shallow');
-            }
-            var ctx = new_leapp_call(scan_args);
+
+            var ctx = new_leapp_call(['port-inspect', {
+                target_host: ip_addr,
+                shallow_scan: shallow,
+                port_range: ports,
+            }]);
+
             ctx.log(`Scanning ports on ${ip_addr}`);
             ctx.done(callback);
             ctx.run();
             return ctx;
         }
 
         function add_ports_product_info(data) {
-            data = JSON.parse(data);
-            if (typeof(data.ports.tcp) !== undefined) {
+            if (typeof(data.port_scan_result.tcp) !== undefined) {
```

Comment:
the result of this condition will be always false, typeof will return string ("undefined"), so "undefined" !== undefined